File: c dsa codes/arrayadt.c
#include<stdio.h>
#include<stdlib.h>
struct myarray{
    int total_size;
    int used_size;
    int *ptr;
};
void createarray(struct myarray *a,int tsize, int usize){
    (*a).total_size=tsize;
    (*a).used_size=usize;
    (*a).ptr=(int*)malloc(tsize*sizeof(int));
}

void showarray(struct myarray *a){
    for(int i=0;i<(*a).used_size;i++){
        printf("%d\n",(*a).ptr[i]);
    }
}

void setvalue(struct myarray *a){
    for(int i=0;i<(*a).used_size;i++){
        int n;
        scanf("%d",&n);
        (*a).ptr[i]=n;
    }
}

int main(){
    struct myarray marks;
    createarray(&marks,100,2);
    setvalue(&marks);
    showarray(&marks);
    return 0;
}

File: java dsa codes/arrayadt.java
import java.util.*;

class myarray {
    int total_size;
    int used_size;
    int[] ptr;

    void createarray(int tsize, int usize) {
        total_size = tsize;
        used_size = usize;
        ptr = new int[tsize];
    }

    void showarray() {
        for (int i = 0; i < used_size; i++) {
            System.out.println(ptr[i]);
        }
    }

    void setvalue() {
        Scanner sc = new Scanner(System.in);
        for (int i = 0; i < used_size; i++) {
            System.out.println("enter the number");

            int n = sc.nextInt();
            ptr[i] = n;

        }
        sc.close();
    }
}

public class arrayadt {
    public static void main(String[] arg) {
        myarray marks = new myarray();
        marks.createarray(100, 2);
        marks.setvalue();
        marks.showarray();
    }
}
